pkg/proxy/dialer.go

```diff
@@ -66,6 +81,11 @@ func ciliumDialer(identity int, network, address string) (net.Conn, error) {
 		return nil, fmt.Errorf("unable to connect: %s", err)
 	}
 
+	if err := setKeepAlive(c); err != nil {
```

Comment:
Shouldn't we also call `TCPConn.SetLinger` with a delay consistent with `proxyConnectionCloseTimeout`, in order to bound the time it takes to close a connection?
Again, we'd need to do that on request connections as well.